frontend/ui/translator.lua

```diff
@@ -477,31 +505,34 @@ Show translated text in TextViewer, with alternate translations
 @string text
 @string target_lang[opt] (`"en"`, `"fr"`, `…`)
 @string source_lang[opt="auto"] (`"en"`, `"fr"`, `…`) or `"auto"` to auto-detect source language
+@bool translate_page "true" to use book language as source language, show main translation only, bigger output window
 --]]
-function Translator:showTranslation(text, target_lang, source_lang, from_highlight, page, index)
+function Translator:showTranslation(text, target_lang, source_lang, from_highlight, page, index, translate_page)
     if Device:hasClipboard() then
         Device.input.setClipboardText(text)
     end
 
     local NetworkMgr = require("ui/network/manager")
-    if NetworkMgr:willRerunWhenOnline(function() self:showTranslation(text, target_lang, source_lang, from_highlight, page, index) end) then
+    if NetworkMgr:willRerunWhenOnline(function()
+                self:showTranslation(text, target_lang, source_lang, from_highlight, page, index, translate_page)
+            end) then
         return
     end
 
     -- Wrap next function with Trapper to be able to interrupt
     -- translation service query.
     local Trapper = require("ui/trapper")
     Trapper:wrap(function()
-        self:_showTranslation(text, target_lang, source_lang, from_highlight, page, index)
+        self:_showTranslation(text, target_lang, source_lang, from_highlight, page, index, translate_page)
     end)
 end
 
-function Translator:_showTranslation(text, target_lang, source_lang, from_highlight, page, index)
+function Translator:_showTranslation(text, target_lang, source_lang, from_highlight, page, index, translate_page)
```

Comment:
Just some thoughts (not insisting too much, but just in case you'd agree :):
`translate_page` still feels a bit odd.
Translator can be seen as some utility tool, and shouldn't really be that linked to the idea of translating a _book_ page or a fragment of a _book_ content. It could be used to translate a Book description, or some Wikipedia result content.
So, ideally, the params shouldn't indicate what's the source (with some inner logic to decide what's expected as the result), but directly what's expected. So, "detailed" vs "simple".
We do pass from_highlight/page/index, so yes, there is a bit of outer context, but that's just for adding buttons.
(Isn't from_highlight a synonym of translate_fragment, ie. it's not there when you translate a page? Or can we don't have it even with fragments?)